types/proposal.go

```diff
@@ -24,25 +24,35 @@ var (
 // a so-called Proof-of-Lock (POL) round, as noted in the POLRound.
 // If POLRound >= 0, then BlockID corresponds to the block that is locked in POLRound.
 type Proposal struct {
-	Type      tmproto.SignedMsgType
-	Height    int64     `json:"height,string"`
-	Round     int32     `json:"round"`     // there can not be greater than 2_147_483_647 rounds
-	POLRound  int32     `json:"pol_round"` // -1 if null.
-	BlockID   BlockID   `json:"block_id"`
-	Timestamp time.Time `json:"timestamp"`
-	Signature []byte    `json:"signature"`
+	Type            tmproto.SignedMsgType
+	Height          int64     `json:"height,string"`
+	Round           int32     `json:"round"`     // there can not be greater than 2_147_483_647 rounds
+	POLRound        int32     `json:"pol_round"` // -1 if null.
+	BlockID         BlockID   `json:"block_id"`
+	Timestamp       time.Time `json:"timestamp"`
+	Signature       []byte    `json:"signature"`
+	TxKeys          []TxKey   `json:"tx_keys"`
+	Header          `json:"header"`
+	LastCommit      *Commit      `json:"last_commit"`
+	Evidence        EvidenceList `json:"evidence"`
+	ProposerAddress Address      `json:"proposer_address"` // original proposer of the block
```

Comment:
curiosity: what's this for? I'm unable to track it down.

Reply:
good point! removed